codespell_lib/tests/test_dictionary.py

```diff
@@ -87,13 +84,13 @@ def _check_aspell(phrase, msg, in_aspell, fname, languages):
 
 
 def _check_err_rep(err, rep, in_aspell, fname, languages):
-    assert ws.match(err) is None, 'error %r has whitespace' % err
-    assert comma.match(err) is None, 'error %r has a comma' % err
+    assert not re.search(r'\s', err), 'error %r has whitespace' % err
```

Comment:
Have you benchmarked this? There's a rather unclear note in the docs:
https://docs.python.org/3/library/re.html#re.compile

But if this is being processed from scratch each time it will do it for each match in each line (>32,000 lines alone)...

Reply:
And there's a bunch of other regexes that are not optimized:
```python
    for (r, msg) in [
            (r'^,',
             '%s starts with a comma'),
            (r'\s,',
             '%s contains a whitespace character followed by a comma'),
            (r',\s\s',
             '%s contains a comma followed by multiple whitespace characters'),
            (r',[^ ]',
             '%s contains a comma *not* followed by a space'),
            (r'\s+$',
             '%s has a trailing space'),
            (r'^[^,]*,\s*$',
             '%s has a single entry but contains a trailing comma')]:
        assert not re.search(r, rep), (msg % (prefix,))
```
Honestly, I don't see how optimizing the first regex helps in any way. EIther we optimize all regexes, or we optimzie none.